module/VuFind/src/VuFind/Db/Service/LoginTokenService.php

```diff
@@ -173,6 +236,19 @@ public function getBySeries(string $series): array
      */
     public function deleteExpired(DateTime $dateLimit, ?int $limit = null): int
     {
-        return $this->getDbTable('LoginToken')->deleteExpired($dateLimit->format('Y-m-d H:i:s'), $limit);
+        // Date limit ignored since login token already contains an expiration time.
+        $subQueryBuilder = $this->entityManager->createQueryBuilder();
+        $subQueryBuilder->select('lt.id')
+            ->from($this->getEntityClass(LoginTokenEntityInterface::class), 'lt')
+            ->where('lt.expires < :dateLimit')
+            ->setParameter('dateLimit', $dateLimit->getTimestamp());
+        if ($limit) {
+            $subQueryBuilder->setMaxResults($limit);
+        }
+        $queryBuilder = $this->entityManager->createQueryBuilder();
+        $queryBuilder->delete($this->getEntityClass(LoginTokenEntityInterface::class), 'lt')
+            ->where('lt.id IN (:tokens)')
+            ->setParameter('tokens', value: $subQueryBuilder->getQuery()->getResult());
```

Comment:
I don't think there's a need to use a named parameter here:

```suggestion
            ->setParameter('tokens', $subQueryBuilder->getQuery()->getResult());
```